src/PoolManager.sol

```diff
@@ -104,7 +103,10 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
         override
         returns (int24 tick)
     {
-        if (key.fee.isStaticFeeTooLarge()) revert FeeTooLarge();
+        // For dynamic fee pools, the initial fee is read from the lower bits of the fee parameter
+        uint24 swapFee = key.fee.getSwapFee();
+        swapFee.validateSwapFee();
+        if (key.fee.isDynamicFee()) key.fee = SwapFeeLibrary.DYNAMIC_FEE_FLAG;
```

Comment:
This seems odd? The key.fee already has information about whether or not it is a DYNAMIC_FEE.

Reply:
Ah interesting - This is set because you allow the lower bits of the key to be the initial dynamic fee. I would actually lean against this in my opinion. The `key` of a pool should only contain data that is unchanging because it is used to find unique pools and to submit a key with information that actually will be changed and actually is not part of the key used to initialize a pool is not good practice imo. I understand this may save gas but I think its quite confusing and not how this data should be used. 

It also makes this line of code very odd (hence my confusion in the comment above).